modules/core/include/opencv2/core/hal/intrin.hpp

```diff
@@ -346,9 +298,15 @@ namespace CV__SIMD_NAMESPACE {
     typedef v_uint64x4   v_uint64;
     typedef v_int64x4    v_int64;
     typedef v_float32x8  v_float32;
+    typedef v_mask8x32   v_mask8;
+    typedef v_mask16x16  v_mask16;
+    typedef v_mask32x8   v_mask32;
+    typedef v_maskf32x8  v_maskf32;
```

Comment:
Is it necessary to retain special mask types for floating point types? AFAIK for x86 and NEON integers and floats use the same registers so type casting is used for syntactic reasons only.

Reply:
No not necessary, actually the reason behind these special types is just because I was worried too much about the plenty of switching between integer and floating-point unit domains in old x86 processors which may occur delays but I guess compilers can handle it, so I think we can just reinterpreting the return of float-point comparisons to integer types.